src/xunit.netcore.extensions/Discoverers/ConditionalFactDiscoverer.cs

```diff
@@ -23,45 +24,71 @@ public ConditionalFactDiscoverer(IMessageSink diagnosticMessageSink) : base(diag
+        }
 
-            string conditionMemberName = factAttribute.GetConstructorArguments().FirstOrDefault() as string;
-            Type declaringType = testMethodInfo.DeclaringType;
-            string[] symbols = conditionMemberName.Split('.');
+        // This helper method evaluates the given condition member names for a given set of test cases.
+        // If any condition member evaluates to 'false', the test cases are marked to be skipped.
+        // The skip reason is the collection of all the condition members that evalated to 'false'.
+        internal static IEnumerable<IXunitTestCase> DiscoverConditionalTestCases(
+                                                        ITestFrameworkDiscoveryOptions discoveryOptions,
+                                                        IMessageSink diagnosticMessageSink,
+                                                        ITestMethod testMethod,
+                                                        IEnumerable<IXunitTestCase> testCases,
+                                                        IEnumerable<string> conditionMemberNames)
+        {
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            Type testMethodDeclaringType = testMethodInfo.DeclaringType;
+            List<string> falseConditions = new List<string>();
 
-            if (symbols.Length == 2)
+            foreach (string entry in conditionMemberNames)
             {
-                conditionMemberName = symbols[1];
-                ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
-                if (type != null)
+                string conditionMemberName = entry;
+                string[] symbols = conditionMemberName.Split('.');
+                Type declaringType = testMethodDeclaringType;
+
+                if (symbols.Length == 2)
                 {
-                    declaringType = type.ToRuntimeType();
+                    conditionMemberName = symbols[1];
+                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
+                    if (type != null)
+                    {
+                        declaringType = type.ToRuntimeType();
+                    }
                 }
-            }
-            
-            MethodInfo conditionMethodInfo;
-            if (conditionMemberName == null ||
-                (conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
-            {
-                return new[] {
+
+                MethodInfo conditionMethodInfo;
+                if (conditionMemberName == null ||
+                    (conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
+                {
+                    return new[] {
```

Comment:
Since the close brace isn't also on this line, the open brace should be on the next line.